rpcs3/rpcs3qt/cheat_manager.cpp

```diff
@@ -5,1068 +5,1012 @@
 		{
-			QMessageBox::warning(this, tr("Error applying value"), tr("Couldn't patch memory"), QMessageBox::Ok);
-			return;
+			auto* child = parent->child(index);
+			if (child->text(0) == QString::fromStdString(cheat_name))
+			{
+				return child;
+			}
 		}
-	});
 
-	// Search UI
-	connect(btn_new_search, &QPushButton::clicked, [this](bool /*checked*/)
-	{
-		offsets_found.clear();
-		do_the_search();
-	});
+		log_cheat.error("Failed to find the cheat '%s' for game '%s'", cheat_name, game_name);
+		return nullptr;
+	};
 
-	connect(edt_cheat_search_value, &QLineEdit::textChanged, this, [btn_new_search, this](const QString& text)
+	const auto& constant_cheats = g_cheat_engine.get_active_constant_cheats();
+	for (const auto& cheat : constant_cheats)
 	{
-		if (btn_new_search)
-		{
-			btn_new_search->setEnabled(!text.isEmpty());
-		}
-		if (btn_filter_results)
+		const auto name = cheat.get_name();
+		auto* node      = find_node(name.first, name.second);
+
+		if (!node)
 		{
-			btn_filter_results->setEnabled(!text.isEmpty() && !offsets_found.empty());
+			continue;
 		}
-	});
 
-	connect(btn_filter_results, &QPushButton::clicked, [this](bool /*checked*/) { do_the_search(); });
+		node->setCheckState(0, Qt::CheckState::Checked);
+	}
 
-	connect(lst_search, &QListWidget::customContextMenuRequested, [this](const QPoint& loc)
+	const auto& queued_cheats = g_cheat_engine.get_queued_cheats();
+	for (const auto& cheat : queued_cheats)
 	{
-		const QPoint globalPos = lst_search->mapToGlobal(loc);
-		const int current_row  = lst_search->currentRow();
-		QListWidgetItem* item  = lst_search->item(current_row);
+		const auto name = cheat.get_name();
+		auto* node      = find_node(name.first, name.second);
 
-		// Skip if the item was a placeholder
-		if (!item || item->data(Qt::UserRole).toBool())
-			return;
+		if (!node)
+		{
+			continue;
+		}
 
-		QMenu* menu = new QMenu();
+		node->setBackground(0, QBrush(Qt::darkGreen));
+	}
+}
 
-		QAction* add_to_cheat_list = new QAction(tr("Add to cheat list"), menu);
+void cheat_manager_dialog::filter_cheats(const QString& game_id, const QString& term)
+{
+	QStringList words;
 
-		const u32 offset       = offsets_found[current_row];
-		const cheat_type type  = static_cast<cheat_type>(cbx_cheat_search_type->currentIndex());
+	const auto str_words = fmt::split(term.toStdString(), {" "});
+	for (const auto& str : str_words)
+	{
+		words << QString::fromStdString(str);
+	}
 
-		connect(add_to_cheat_list, &QAction::triggered, [name = Emu.GetTitle(), offset, type, this]()
-		{
-			if (g_cheat.exist(name, offset))
-			{
-				if (QMessageBox::question(this, tr("Cheat already exist"), tr("Do you want to overwrite the existing cheat?"), QMessageBox::Ok | QMessageBox::Cancel) != QMessageBox::Ok)
-					return;
-			}
+	const auto* root = m_tree->invisibleRootItem();
 
-			std::string comment;
-			if (!cheat_engine::is_addr_safe(offset))
-				comment = "Unsafe";
+	for (auto i = 0; i < root->childCount(); i++)
```

Comment:
please use int, it's kinda dumb to read auto everywhere.
```suggestion
	for (int i = 0; i < root->childCount(); i++)
```